spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Update installation widget."""
+
+# Standard library imports
+import sys
+import os
+import subprocess
+from urllib.request import urlretrieve
+from tempfile import TemporaryDirectory
+import threading
+
+# Third-party imports
+from qtpy.QtCore import QEvent, QObject, Qt, Signal
+from qtpy.QtWidgets import (QDialog, QHBoxLayout, QMessageBox,
+                            QLabel, QProgressBar, QPushButton, QVBoxLayout,
+                            QWidget)
+
+# Local imports
+from spyder import __version__
+from spyder.config.base import _
+from spyder.utils.icon_manager import ima
+from spyder.utils.palette import QStylePalette
+from spyder.utils.programs import is_module_installed
+
+
+# Update installation process statuses
+NO_STATUS = __version__
+DOWNLOADING_INSTALLER = _("Downloading installer")
+INSTALLING = _("Installing")
+FINISHED = _("Installation finished")
+PENDING = _("Pending update")
+CHECKING = _("Checking for updates")
+CANCELLED = _("Cancelled")
+
+
+class UpdateInstallation(QWidget):
+    """Update progress installation widget."""
+
+    def __init__(self, parent):
+        super(UpdateInstallation, self).__init__(parent)
```

Comment:
```suggestion
        super().__init__(parent)
```

That was the old way (i.e. Python 2 compatible) of using `__init__`, but it's no longer necessary.